backend/src/main/java/hu/flowacademy/worksheet/service/WorksheetService.java

```diff
@@ -35,24 +34,24 @@
 
     private final WorksheetRepository worksheetRepository;
     private final PagingProperties pagingProperties;
-    private final PartnerRepository partnerRepository;
+    private final PartnerService partnerService;
+    private final UserService userService;
 
     public Worksheet saveWorksheet(@NonNull WorksheetDTO worksheetDTO) throws ValidationException {
         Worksheet worksheet = buildWorksheet(worksheetDTO);
         validateWorksheet(worksheet);
-        if (worksheet.getWorksheetStatus() != WorksheetStatus.REPORTED) {
-            worksheet.setWorksheetStatus(WorksheetStatus.CREATED);
-        }
-        worksheet.setCreatedAtRealTime(LocalDateTime.now());
-        return worksheetRepository.save(worksheet);
+        return worksheetRepository.save(worksheet.toBuilder()
+                .worksheetStatus(worksheet.getWorksheetStatus() != WorksheetStatus.REPORTED ? WorksheetStatus.CREATED
```

Comment:
Ha created v closed akkor created, különben a reported? Closed-ra nem ezzel kell állítani?

Reply:
Szia!
Van egy "munkalap mentés", meg egy "munkalap mentés és készre jelentés" gomb létrehozásnál, azokat próbálja ez a rész kezelni.

Átírtam a 45. sort, h ne getter-rel kérje el a beállítást, hanem fixen reportedra tegye, és akkor következetesebb talán.